### PR TITLE
enable to include functions versus exclude

### PR DESCRIPTION
**Pull request by kamaz**
In some cases, we found an exclude option a little bit awkward especially when you have only a single function to apply Layer on. 

I've added an include option which works as exclude but instead includes functions.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -83,6 +83,15 @@ export default class NewRelicLambdaLayerPlugin {
       return;
     }
 
+    const { exclude = [] } = this.config;
+    const { include } = this.config;
```
Comment (mrickard) [inline src/index.ts]: I'd suggest that include and exclude should have identical structures to reduce any future confusion.  So 
``` const { include = [], exclude = [] } = this.config ```

Comment (kamaz) [reply src/index.ts]: good point aligned the functionality

Comment (mrickard) [reply src/index.ts]: And then that could be carried throughout the file where that's used.

File: src/index.ts [hunk 2]
```diff
@@ -194,6 +202,18 @@ export default class NewRelicLambdaLayerPlugin {
...
     }
 
+    const { include = [] } = this.config;
+    if (
+      !_.isEmpty(include) &&
+      _.isArray(include) &&
+      include.indexOf(funcName) === -1
+    ) {
+      this.serverless.cli.log(
+        `Excluded function ${funcName}; is not part of include skipping`
```
Comment (mrickard) [inline src/index.ts]: I'd suggest ```Function ${funcName} is not included for instrumentation; skipping.```

File: src/index.ts (hunk 1 above)
Comment (kamaz) [reply src/index.ts]: they are different methods which means they would need to be passed through as arguments

but I've noticed that include was missing in subscription (added now)

File: src/index.ts (hunk 2 above)
Comment (kamaz) [reply src/index.ts]: updated the text

File: src/index.ts (hunk 1 above)
Comment (kamaz) [reply src/index.ts]: I think you meant defaulting to an empty array. Yes, I've corrected that as part of fix so it follows the same approach.

Comment (mrickard) [reply src/index.ts]: Yes! Apologies for the shorthand there.

Comment (kamaz) [reply src/index.ts]: cool I've extracted to common function and used the shorthand 

looks better 😄

File: src/index.ts [hunk 3]
```diff
@@ -194,10 +202,8 @@ export default class NewRelicLambdaLayerPlugin {
       return;
     }
 
-    const { exclude = [] } = this.config;
-    if (_.isArray(exclude) && exclude.indexOf(funcName) !== -1) {
-      this.serverless.cli.log(`Excluded function ${funcName}; skipping`);
-      return;
+    if (this.shouldSkip(funcName)) {
+      return
```
Comment (mrickard) [inline src/index.ts]: We may need a semicolon on this line.  (I have an auto-linting PR that should be merged shortly.)

**Comment by mrickard** [comment]
@kamaz We've merged the enforced-linting branch, so we'll want to rebase this PR against master and resolve the lock conflicts.  (I'm happy to do that.)

**Comment by mrickard** [comment]
tslint output: 
```
ERROR: src/index.ts:129:1 - Delete `⏎`
ERROR: src/index.ts:131:15 - Insert `;`
ERROR: src/index.ts:206:13 - Insert `;`
ERROR: src/index.ts:252:1 - Replace `········` with `······`
ERROR: src/index.ts:253:1 - Delete `··`
ERROR: src/index.ts:283:17 - Insert `;`
```

**Comment by kamaz** [comment]
no worries I will merge with the latest and push a new commit

**Comment by kamaz** [comment]
@mrickard all done

**Comment by mrickard** [comment]
@kamaz Excellent! Looking now...

**Pull request merged**